src/EditorFeatures/CSharp/DecompiledSource/CSharpDecompiledSourceService.cs

```diff
@@ -103,14 +103,26 @@ public static async Task<Document> FormatDocumentAsync(Document document, Syntax
 
             var fullTypeName = new FullTypeName(fullName);
 
-            // Try to decompile; if an exception is thrown the caller will handle it
-            var text = decompiler.DecompileTypeAsString(fullTypeName);
+            try
+            {
+                var text = decompiler.DecompileTypeAsString(fullTypeName);
+
+                text += "#if false // " + CSharpEditorResources.Decompilation_log + Environment.NewLine;
+                text += logger.ToString();
+                text += "#endif" + Environment.NewLine;
 
-            text += "#if false // " + CSharpEditorResources.Decompilation_log + Environment.NewLine;
-            text += logger.ToString();
-            text += "#endif" + Environment.NewLine;
+                return document.WithText(SourceText.From(text));
+            }
+            catch (NotSupportedException)
+            {
+                // IL Spy throws for "Decompiling types that are not part of the main module is not supported."
+            }
+            catch (InvalidOperationException)
+            {
+                // IL Spy throws if it can't find the type name in its type system.
```

Comment:
Why would this happen though? Wouldn't this indicate a bug on our side?

Reply:
No idea, but given we have a fallback for creating source from metadata, I don't think the NFW is helping us in any way regardless.